slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -104,6 +106,9 @@ class SLAMServiceImpl final : public SLAMService::Service {
     double translation_weight = 10.0;
     double rotation_weight = 1.0;
 
+    std::mutex map_builder_mutex;
+    mapping::MapBuilder map_builder;
```

Comment:
It's preferable to have a mutex and any variables it's protecting be private. This makes it easier to check that the mutex is locked before accessing the variables.

Reply:
Why/how does it make it easier to check that the mutex is locked before accessing the variables when it is private?

Initially, I had it private but then ran into problems when I was using it. I'll take another look to see if I can make it private after all.

Reply:
If the mutex and the variables it protects are private, then we only need to read the implementation of the class in order to check that the mutex is locked before accessing the variables. Otherwise, we have to look for usages of the mutex and variables outside of the class, so there's just a lot more to verify. Keeping the mutex private also makes it easier to reason about mutex acquisition order.

Reply:
Ah, right, regarding why I moved it to be public:

Having it private caused issues when I am calling `slamService.map_builder...` in the `slam_service_test.cc`, like this:

```cpp
void checkCartoMapBuilderParameters(SLAMServiceImpl& slamService) {
    auto pose_graph_options =
        slamService.map_builder.map_builder_options_.pose_graph_options();
...
```

resulting in this error:

```bash
/Users/kat/slam/slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc:13:21: error: 'map_builder' is a private member of 'viam::SLAMServiceImpl'
        slamService.map_builder.map_builder_options_.pose_graph_options();
```

Do you have suggestions on how I can resolve this elegantly? 

Something that comes to my mind is that I could implement functions within the `slamService` that retrieve & return those parameters from `map_builder`. Would this be the way to go here? Or are there alternative solutions?